src/Functional/FromEntries.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * @package   Functional-php
+ * @author    Hugo Sales <hugo@hsal.es>
+ * @copyright 2021 Lars Strojny
+ * @license   https://opensource.org/licenses/MIT MIT
+ * @link      https://github.com/lstrojny/functional-php
+ */
+
+namespace Functional;
+
+use Functional\Exceptions\InvalidArgumentException;
+use Traversable;
+
+/**
+ * Inspired by JavaScript's `Object.from_entries`,
+ * convert an array of key-value tuples into a key-value map
+ *
+ * @see Functional\entries
+ * @param Traversable|array $collection
+ * @return array
+ * @no-named-arguments
+ */
+function from_entries($collection)
+{
+    InvalidArgumentException::assertCollection($collection, __FUNCTION__, 1);
+
+    $aggregation = [];
+    foreach ($collection as [$key, $value]) {
```

Comment:
Let's throw an exception if it's not a pair

Reply:
You mean if collection doesn't have string keys?

Reply:
Right, reading this again I see what you mean. Makes sense, but I'm not sure how to check that

Reply:
Something like this:
```
foreach ($collection as $enty) {
     InvalidArgumentException::assertPair($entry);
}
```

```
public static function assertPair($pair): void
{
    if (!(is_array($pair) || $pair instanceof ArrayAccess) && !isset($pair[0], $pair[1])) {
          … throw
    }
}
```

Then add a few test cases for the `assertPair()` method